imgproc.go

```diff
@@ -1930,3 +1931,164 @@ func PhaseCorrelate(src1, src2, window Mat) (phaseShift Point2f, response float6
 		Y: float32(result.y),
 	}, float64(responseDouble)
 }
+
+// ToImage converts a Mat to a image.Image.
+func (m *Mat) ToImage() (image.Image, error) {
+	t := m.Type()
+	if t != MatTypeCV8UC1 && t != MatTypeCV8UC3 && t != MatTypeCV8UC4 {
+		return nil, errors.New("ToImage supports only MatType CV8UC1, CV8UC3 and CV8UC4")
+	}
+
+	width := m.Cols()
+	height := m.Rows()
+
+	if MatTypeCV8UC1 == t {
+		img := image.NewGray(image.Rect(0, 0, width, height))
+		q := ([]byte)(m.DataPtrUint8())
+		copy(img.Pix, q[0:])
+		return img, nil
+	}
+
+	// MatTypeCV8UC3 and MatTypeCV8UC4
+	dst := NewMat()
+	if MatTypeCV8UC3 == t {
```

Comment:
If you add a `defer dst.Close()` here it will fix the memory leak complained about in https://ci.appveyor.com/project/deadprogram/gocv/builds/35772719#L626